src/package/packageCreate.ts

```diff
@@ -0,0 +1,145 @@
+>;
+
+export function _createPackage2RequestFromContext(project: SfProject, options: PackageCreateOptions): Package2Request {
+  const namespace = options.noNamespace ? '' : project.getSfProjectJson().getContents().namespace || '';
+  return {
+    Name: options.name,
+    Description: options.description,
+    NamespacePrefix: namespace,
+    ContainerOptions: options.packageType,
+    IsOrgDependent: options.orgDependent,
+    PackageErrorUsername: options.errorNotificationUsername,
+  };
+}
+
+/**
+ * Generate packageDirectory json entry for this package that can be written to sfdx-project.json
+ *
+ * @param project
+ * @param packageId the 0Ho id of the package to create the entry for
+ * @private
+ */
+
+export function _generatePackageDirEntry(
+  project: SfProject,
+  options: PackageCreateOptions
+): PackageDir[] | NamedPackageDir[] {
+  let packageDirs: NamedPackageDir[] = project.getPackageDirectories();
+  if (!packageDirs) {
+    packageDirs = [];
+  }
+
+  // add an entry if it doesn't exist
+  // or update an existing entry if it matches path but has no package or id attribute (W-5092620)
+  let packageDir: NamedPackageDir =
+    project.getPackage(options.name) ||
+    project
+      .getPackageDirectories()
+      // TODO: I don't understand where id is coming from, because it is not in sfdx-project.json schema
+      .find((pd: NamedPackageDir & { id: string }) => pd.path === options.path && !pd.id && !pd.package);
+
+  if (packageDir) {
+    // update existing entry
+    packageDir.package = options.name;
+    packageDir.versionName ??= pkgUtils.DEFAULT_PACKAGE_DIR.versionName;
+    packageDir.versionNumber ??= pkgUtils.DEFAULT_PACKAGE_DIR.versionNumber;
+    // set as default if this is the only entry or no other entry is the default
+    if (!Reflect.getOwnPropertyDescriptor(packageDir, 'default')) {
+      packageDir.default = !pkgUtils.getConfigPackageDirectory(packageDirs, 'default', true);
+    }
+  } else {
+    // add new entry
+    packageDir = pkgUtils.DEFAULT_PACKAGE_DIR as NamedPackageDir;
+    packageDir.package = options.name;
+    // set as default if this is the only entry or no other entry is the default
+    packageDir.default = !pkgUtils.getConfigPackageDirectory(packageDirs, 'default', true);
+    packageDir.path = options.path;
+
+    packageDirs.push(packageDir);
+  }
+
+  return packageDirs;
+}
+
+/**
+ * Generate package alias json entry for this package that can be written to sfdx-project.json
+ *
+ * @param context
+ * @param packageId the 0Ho id of the package to create the alias entry for
+ * @private
+ */
+export function _generatePackageAliasEntry(
+  project: SfProject,
+  options: PackageCreateOptions,
+  packageId: string
+): { [key: string]: string } {
+  const packageAliases = project.getSfProjectJson().getContents().packageAliases || {};
+
+  const packageName = options.name;
+  packageAliases[packageName] = packageId;
+
+  return packageAliases;
+}
+
+export async function createPackage(
```

Comment:
I don't see the `org` parameter used, if it is used, you can remove the `connection` parameter and just get it from `org.getConnection()`

Reply:
Actually I would prefer to pass connection rather than org. I will look at removing it.